### ISSUE TITLE
Webpack Build Errors with Yarn v3 and `supports-color@^9`

### ISSUE THREAD

**Issue by ChrisSargent** [gatsbyjs/gatsby#34101]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Gatsby has a dependency on `chalk` which also depends on `supports-color`. The latest version of `supports-color` uses ESM modules and uses an import syntax like this:

```js
import process from 'node:process';
import os from 'node:os';
import tty from 'node:tty';
```

In my case, I am using yarn v3 (the error does not reproduce with v1) and had installed the latest version of `lint-staged` which depends on the latest ESM-enabled version of `supports-color`.

This meant that Gatsby was failing when using `yarn build` (not `yarn start`) at the end of the build, presumably when it was trying to output something to the console.

The errors are described below.

In my case, I unblocked the error by setting a resolution in my `package.json` file like:

```json
"resolutions": {
  "supports-color": "^8",
}
```

Posting this here to hopefully help someone else save the 2 days of debugging this took to find (was in a pretty large monorepo and a commit linting tool update was the last place we looked!) and also hoping that perhaps the Gatsby Webpack config can be updated to solve this issue in future as more packages start depending on later versions of `supports-color`

Related: https://stackoverflow.com/questions/69864315/webpack-esm-unable-to-resolve-os

### Reproduction Link

https://github.com/ChrisSargent/webpack-gatsby-issue

### Steps to Reproduce

1. Run `yarn install`
2. Run `yarn build`

### Expected Result

The build completes without errors

### Actual Result

The build has errors as described below:

```
ERROR #98123  WEBPACK

Generating SSR bundle failed

Reading from "node:os" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs.

File: node:os

ERROR #98123  WEBPACK

Generating SSR bundle failed

Reading from "node:process" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs.

File: node:process

ERROR #98123  WEBPACK

Generating SSR bundle failed

Reading from "node:tty" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs.

File: node:tty
```

### Environment

```shell
System:
    OS: macOS 11.5.2
    CPU: (4) x64 Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.18.1 - /private/var/folders/dw/lgl4dq6j3r30y378tlykv85r0000gn/T/xfs-e6b26d30/node
    Yarn: 3.1.1 - /private/var/folders/dw/lgl4dq6j3r30y378tlykv85r0000gn/T/xfs-e6b26d30/yarn
    npm: 6.14.15 - ~/.nvm/versions/node/v14.18.1/bin/npm
  Languages:
    Python: 3.9.7 - /usr/local/opt/python/libexec/bin/python
  Browsers:
    Chrome: 96.0.4664.55
    Firefox: 93.0
    Safari: 14.1.2
  npmPackages:
    gatsby: ^3.11.1 => 3.14.6
```

**Comment by ChrisSargent** [comment]
Bump

**Comment by aaronadamsCA** [comment]
I just saw this in the wild for the first time, on a failed Gatsby Cloud build. We have multiple sites in the same monorepo, and only some are failing.

I think I've worked around the issue with resolution `"lint-staged/supports-color": "^8"`, thanks a ton @ChrisSargent for isolating this. We would've likely been stuck for a while. Unfortunately Yarn 3 builds on Gatsby Cloud just stopped working, so I'll have to confirm once that's fixed.

**Comment by chkmrv** [comment]
I reproduce this error with yarn 1.22.17
I am updating to gatsby 4 and webpack 5 and have the same error 
![image](https://user-images.githubusercontent.com/3708844/166493400-b37116d0-3a03-46cb-8a36-01940d7cf1b8.png)

**Comment by chkmrv** [comment]
I hope it will help someone.

I downgrade to this:
"dotenv": "^10.0.0",
"gatsby": "3.14.6",

And run **yarn add webpack@latest --legacy-peer-deps**

**Comment by wardpeet** [comment]
We don't support esm today that's why you're seeing this error.

**Comment by yangbooom** [comment]
Great thanks to @ChrisSargent  and @aaronadamsCA.

I also encountered the issue with `yarn v3` and monorepo.
To elaborate the solution, add ```"resolutions": {
  "supports-color": "^8",
}```
to the `pacakge.json` **that is in the root of your monorepo**, NOT of ones in the `packages` directory.
